[website] Endringslogg

HalvorHaugan opened the pull request:
WIP

HalvorHaugan committed:
sidebar entry

stainawarjar committed:
Add sidebar entry and basic presentation page

stainawarjar committed:
Add unwired search and filters

stainawarjar committed:
Update sanity schema

* add slug type for endringslogg
* update schema to reflect current design

stainawarjar committed:
Update sidebar entry to standalone link

stainawarjar committed:
Add presentation of changelog content

stainawarjar committed:
Make month-headers sticky

stainawarjar committed:
Merge branch 'main' into endringslogg

stainawarjar committed:
Reduce padding under month-header and use aksel brand tokens

stainawarjar committed:
Add month-header gradient transparency

stainawarjar committed:
Add changelog page to render individual log entries

stainawarjar committed:
Bump headings to next heading-elvel for changelog list

stainawarjar committed:
Fix error when article lacks 'innhold'

stainawarjar committed:
Fix log nesting and clean up styling

stainawarjar committed:
Clean up styles

stainawarjar committed:
Add rudimentary show more functionality

stainawarjar committed:
Try out fixes for various issues

stainawarjar committed:
Add rudimentary filtering (not search)

stainawarjar committed:
Make filter chips use next/navigation

stainawarjar committed:
Finish basic year and category filtering

stainawarjar committed:
Filter log entries by search string

stainawarjar committed:
Add changelog entries to global search

stainawarjar committed:
Make sanity logentry validations more robust

If a validation error was introduced through
any of the fields organized under the ´fremhevet´-flag,
if the ´fremhevet´-flag was deactivated,
the validation errors persisted, despite no longer being relevant.

These updated validations account for whether ´fremhevet´ is set or not.

stainawarjar committed:
Account for possibly missing hero image despite flag set

stainawarjar committed:
Tweak search input and show more collapse

HalvorHaugan committed:
Add show more toggle til schema

HalvorHaugan committed:
Types

stainawarjar committed:
Add dynamic toc and use date-fns for dates

stainawarjar committed:
Set query type to any

stainawarjar committed:
Remove endringslogg from global search

stainawarjar committed:
Fix show more and fremhevet

stainawarjar committed:
Use visMer flag, add demoable pulsing bullet

stainawarjar committed:
Merge branch 'main' into endringslogg

stainawarjar committed:
Adjust collapsed height of show more

stainawarjar committed:
Add toc to article page

stainawarjar committed:
Improve search and test log entry bleed on mobile

stainawarjar committed:
Respond immediately when clicking filter chips

stainawarjar committed:
Fix static route generation for `slug`

stainawarjar committed:
Merge branch 'main' into endringslogg

stainawarjar committed:
Add prev and next footer on article page

HalvorHaugan committed:
Merge branch 'main' into endringslogg

HalvorHaugan committed:
adjustments

stainawarjar committed:
Clean up prev/next link cards in endringslogg article

HalvorHaugan committed:
dl

stainawarjar committed:
Improve after-collapse highlight

HalvorHaugan committed:
refactor capitalized

HalvorHaugan committed:
Man må bruke defineQuery for at typer skal bli generert (gjøres med kommando `yarn workspace website update:sanity-schema`)

HalvorHaugan committed:
Fjernet "hardkodede" typer i [slug]/page.tsx

HalvorHaugan committed:
synes dette ser litt ryddigere ut jeg

stainawarjar committed:
Extract ShowMore into separate component

stainawarjar committed:
Remove timeline on mobile and fix circular reference in ShowMore

stainawarjar committed:
Fix height of fremhevet and collapsed (ShowMore)

stainawarjar committed:
Merge branch 'main' into endringslogg

HalvorHaugan committed:
fix Next type error

HalvorHaugan committed:
ignore type error

stainawarjar committed:
Re-use bumpHeadingLevels and move ENDRINGSLOGG_QUERY to queries.ts

HalvorHaugan commented on aksel.nav.no/website/app/_ui/bilde/Bilde.tsx:
```diff
@@ -19,6 +19,8 @@ function Bilde(props: ExtractPortableComponentProps<"bilde">) {
   const imageUrl = urlForImage(props?.value as Image)
     ?.auto("format")
+    // TODO: [endringslogg] Remove before production
+    .dataset("development")
```
Just adding this comment to remind ourselves to remove this.

HalvorHaugan committed:
Remove showChangelogLink

HalvorHaugan committed:
Merge branch 'main' into endringslogg

HalvorHaugan committed:
Remove showChangelogLink from the other page.tsx as well

stainawarjar committed:
Add OG metadat and clean up sanity query types

HalvorHaugan committed:
ChangelogLinkCard changes

HalvorHaugan committed:
avoid reduce and save 3 lines of code

stainawarjar committed:
Fix useOptimistic with wrap in startTransition()

stainawarjar committed:
Move to semantic colors for brand and data-color where possible

stainawarjar committed:
Clean up styles and use data-color where possible

stainawarjar committed:
Fix non-highlighted article heading

HalvorHaugan commented on aksel.nav.no/website/app/dev/(designsystemet)/grunnleggende/endringslogg/_ui/FilterChips.tsx:
```diff
@@ -0,0 +1,96 @@
+    return `${pathname}?${params.toString()}`;
+  }
+
+  const FilterGroup = ({
```
Not a big deal, but I think it's considered bad practice to define a component within a component.

stainawarjar replied:
Indeed it is :blush: Will fix.

HalvorHaugan committed:
adjustments

stainawarjar committed:
Revise sanity queries

HalvorHaugan committed:
misc

HalvorHaugan committed:
comment about h5

HalvorHaugan committed:
Merge branch 'main' into endringslogg

HalvorHaugan committed:
innhold -> content

stainawarjar committed:
Fix persisting search query in search box after page revisit

HalvorHaugan committed:
move toc

stainawarjar committed:
Avoid polluting state in FilterChips

HalvorHaugan committed:
arstall -> årstall (seems to work?), ingen -> alle, misc.

HalvorHaugan committed:
Highlight "Endringslogg" in sidebar when on single article

HalvorHaugan committed:
Merge branch 'main' into endringslogg

HalvorHaugan commented on aksel.nav.no/website/app/dev/(designsystemet)/grunnleggende/endringslogg/_ui/LogEntry.tsx:
```diff
@@ -0,0 +1,219 @@
+                    styles.portableTextFirstHeading,
+                    fremhevet && styles.innholdFremhevet,
+                  )}
+                  data-color={fremhevet ? "aksel-brand-pink" : "neutral"}
```
I don't think the links and the bullets should be gray. Maybe caused by `data-color`?

![image](https://github.com/user-attachments/assets/5c7082c8-5ca6-4089-b77c-d581d7600ad6)

stainawarjar replied:
We should take a closer look at this. It is because of `data-color={"neutral"}`. We can use `data-color={"aksel-brand-blue"}`, but I don't think that gets us the right color. It seems to just use `accent`, which is a different hue to `aksel-brand-blue`. This all has to do with the built-in styling of `CustomPortableText´.

stainawarjar replied:
There was no `aksel-brand-blue`, just `brand-blue`...

stainawarjar committed:
Fix various issues highlighted in review

- Add OG image on changelog page
- Add fallback OG image in changelog entry/article page
- Remove leftover unnecessary background styling
- Simplify styling of timeline
- Treat `year` as `string` instead of `number`
- Use `periode` as denominator in search params instead of `årstall`
- Add `as={NextLink}` to Aksel links
- Use `ReactNode` as type on `children` of `ShowMore.Content`

stainawarjar committed:
Fix type on FilterChips

stainawarjar committed:
Move component definition out of FilterChips

stainawarjar committed:
Use onSubmit instead of onChange in SearchField

HalvorHaugan committed:
Adjustments in slug page

stainawarjar committed:
Fix OG, BoxNew, brand-blue, search button, focusFlash and ShowMore validation

- Update to correct OG image and OG fallback image
- Replace Box.New with BoxNew which supports SSR/"use server"
- Fix data-color="brand-blue"
- Add button to search box
- Modify focusFlash animation on collapse of ShowMore
- Fix type validation of ShowMore children

stainawarjar committed:
Set heading color in article to neutral

stainawarjar committed:
Rework endringslogg-link in sidebar

stainawarjar committed:
Remove todo-comment

KenAJoh commented on aksel.nav.no/website/app/dev/(designsystemet)/grunnleggende/endringslogg/_ui/FilterGroup.tsx:
```diff
@@ -0,0 +1,70 @@
+                } else {
+                  startTransition(() => expect(option));
+                }
+                push(href);
```
Found it a little annoying that the page scrolls while i want to change multiple filters, this should hopefully fix it 👍 
```suggestion
                push(href, {scroll: false});
```

stainawarjar replied:
That did indeed fix it.

KenAJoh commented on aksel.nav.no/website/app/dev/(designsystemet)/grunnleggende/endringslogg/page.tsx:
No idea where this is set, but pink background bleeds around the corners now 
<img width="235" alt="Screenshot 2025-06-27 at 13 20 22" src="https://github.com/user-attachments/assets/967bbb0a-4cd9-4d0b-a44b-272c56331759" />

stainawarjar replied:
Turns out I had added a background color on this particular picture in Sanity. Removing the background color fixes this.

KenAJoh commented on aksel.nav.no/website/app/dev/(designsystemet)/grunnleggende/endringslogg/_ui/SearchField.tsx:
More of a UX-nitpick, but personally when i clear a field (either by hitting Esc, X or just emptying the field manually) i would expect the search to reset. Right now only the two first cases work since they trigger onClear. 
Might be as easy as adding `onClear={(v) => v === "" && resetSearch()}` or something like that.

stainawarjar replied:
Esc and X currently trigger `onClear={() => handleSearch("")}`. When we moved from using `onChange` to `onSubmit` on this one, users have to take action to trigger the update of the search results. Clearing the field + submitting (with Enter or click on search button) does reset the search filter to nothing. We may trigger reset with `onChange={(v) => v === "" && handleSearch("")}`. However, it could be that the user's intention is just to clear the field to type in a different filter, and experience the auto-triggering of a reset as unexpected.
We should probably revisit this search-filter anyway, since the submit-paradigm differs from the two other search interfaces on the same page.
I'll add the onChange, since I think it will match more expectations anyway.

HalvorHaugan replied:
I'm afraid this will feel inconsistent, but we can try it and revisit later.

KenAJoh commented on aksel.nav.no/website/app/dev/(designsystemet)/grunnleggende/endringslogg/[slug]/page.tsx:
Wish i had a shortcut back to the main endringslogg-view from here. Not something that needs to be in v1 ofc
<img width="1063" alt="Screenshot 2025-06-27 at 13 29 29" src="https://github.com/user-attachments/assets/3e464cda-3f50-4615-881e-b4affe4e00c9" />

stainawarjar replied:
This has been discussed, but we should reconsider it for v2

KenAJoh commented on aksel.nav.no/website/app/dev/(designsystemet)/grunnleggende/endringslogg/_ui/Changelog.module.css:
```diff
@@ -0,0 +1,97 @@
+
+.timeline {
+  width: 0;
+  border: 0.5px dashed var(--ax-border-neutral-subtle);
```
This might need to be validated on some different screens since windows often handles half-pixels different mac. So might be 0 on some screens and 1 on others.

HalvorHaugan replied:
Seems to work fine on Windows in both Chrome and Firefox. However, it feels a bit like a hack that could stop working later. Setting it to 1px seems to produce the same result afaics 🤷‍♂️ To be "safe", we could also change to `border-left` or `border-right`.

HalvorHaugan replied:
Looks like 0.5px is rounded up to 1px, so the dotted line is in fact 2px wide. And unless we do something with the dots, it has to be 2px to be centered (1px doesn't add up). So I think I'll just set it to `border: 1px ...`

KenAJoh commented on aksel.nav.no/website/app/dev/(designsystemet)/grunnleggende/endringslogg/[slug]/page.tsx:
```diff
@@ -0,0 +1,176 @@
+  });
+
+  return {
+    title: pageData?.heading,
```
```suggestion
    title: `${pageData?.heading} - Endringslogg`,
```

stainawarjar committed:
Update aksel.nav.no/website/app/dev/(designsystemet)/grunnleggende/endringslogg/_ui/FilterGroup.tsx

Verified to work.

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

stainawarjar committed:
Update page title

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

HalvorHaugan committed:
fix colors

HalvorHaugan committed:
types

HalvorHaugan committed:
Merge branch 'main' into endringslogg

HalvorHaugan committed:
Fix issue where "overflow shadow" would not toggle when ToC changed

stainawarjar committed:
Update search filtering when input field is empty

HalvorHaugan committed:
Merge branch 'main' into endringslogg

HalvorHaugan committed:
round up border width

HalvorHaugan committed:
move toString()

HalvorHaugan committed:
Merge branch 'main' into endringslogg
